socket/sc_sock.c

```diff
@@ -1538,8 +1549,18 @@ int sc_sock_poll_term(struct sc_sock_poll *p)
 		return 0;
 	}
 
+	sc_sock_poll_del(p, &p->wakeup_pipe.fdt, SC_SOCK_READ, NULL);
+	sc_sock_pipe_term(&p->wakeup_pipe);
```

Comment:
maybe we can check error here and then call `sc_sock_poll_set_err()` on error

Reply:
Makes sense.